### PR TITLE
Feature/fix available brands

### PR DESCRIPTION
**Pull request by kathyoungaligent**

### CODE REVIEW

File: app/code/community/Aligent/Emarsys/Model/Filter.php [hunk 1]
```diff
@@ -66,4 +63,24 @@ public function beforeQueryFilter(Varien_Db_Select &$oSelect, $oStore, $aParms)
...
+
+    protected function getAttribute($attrName){
+        $oProductModel = Mage::getModel('catalog/product');
+        $objAttr = Mage::getSingleton('eav/config')->getCollectionAttribute($oProductModel->getResource()->getType(), $attrName);
+        $data = array(
+            'table' => $objAttr->getBackendTable(),
+            'id' => $objAttr->getId()
+        );
+        $objAttr = null;
+        $oProductModel = null;
```
Comment (michaelwylde) [inline app/code/community/Aligent/Emarsys/Model/Filter.php]: could the oProductModel be a static variable that is reused and not thrown away each time?

File: app/code/community/Aligent/Emarsys/Model/Filter.php [hunk 2]
```diff
@@ -66,4 +66,36 @@ public function beforeQueryFilter(Varien_Db_Select &$oSelect, $oStore, $aParms)
 
         Mage::getSingleton('aligent_feeds/log')->log("Catalog Select is: $vSql");
     }
+
+    protected function getAttribute($attrName){
+        if(isset(self::$_attributes[$attrName])) return self::$_attributes[$attrName];
+        $oProductModel = $this->getProductModel();
```
Comment (michaelwylde) [inline app/code/community/Aligent/Emarsys/Model/Filter.php]: Why are you declaring this and not just using it once on line 74?

Comment (kathyoungaligent) [reply app/code/community/Aligent/Emarsys/Model/Filter.php]: Cause line 74 already felt horribly messy.  I guess I can do that though.

File: app/code/community/Aligent/Emarsys/Model/Filter.php [hunk 3]
```diff
@@ -66,4 +66,35 @@ public function beforeQueryFilter(Varien_Db_Select &$oSelect, $oStore, $aParms)
...
+
+        $objAttr = Mage::getSingleton('eav/config')->getCollectionAttribute($this->getProductModel()->getResource()->getType(), $attrName);
+        $data = array(
+            'table' => $objAttr->getBackendTable(),
+            'id' => $objAttr->getId()
+        );
+        self::$_attributes[$attrName] = $data;
+
+        $objAttr = null;
+        $oProductModel = null;
```
Comment (michaelwylde) [inline app/code/community/Aligent/Emarsys/Model/Filter.php]: Should this be removed now?

**Pull request merged**